l1-contracts/src/core/libraries/rollup/ProposeLib.sol

```diff
@@ -70,18 +113,56 @@ library ProposeLib {
   using ChainTipsLib for CompressedChainTips;
 
   /**
-   * @notice  Publishes the body and propose the block
-   * @dev     `eth_log_handlers` rely on this function
+   * @notice  Publishes a new L2 block to the pending chain.
+   * @dev     Handles a proposed L2 block, validates it, and updates rollup state adding it to the pending chain.
+   *          Orchestrates blob validation, header validation, proposer verification, fee calculations, and state
+   *          transitions. Automatically prunes unproven blocks if the proof submission window has passed.
+   *
+   *          Note that some validations and processes are disabled if the chain is configured to run without
+   *          transactions, such as during ignition phase:
+   *          - No fee header computation or L1 gas fee oracle update
+   *          - No inbox message consumption or outbox message insertion
+   *
+   *          Validations performed:
+   *          - Blob commitments against provided blob data: Errors.Rollup__InvalidBlobHash,
+   *            Errors.Rollup__InvalidBlobProof
+   *          - Block header validations (see validateHeader function for details)
+   *          - Proposer signature is valid for designated slot proposer:
+   *            Errors.ValidatorSelection__MissingProposerSignature
+   *          - Inbox hash matches expected value (when txs enabled): Errors.Rollup__InvalidInHash
+   *
+   *          Validations NOT performed:
+   *          - Committee attestations (only proposer signature verified)
+   *          - Transaction validity and state root computation (done at proof submission via a validity proof)
+   *
+   *          State changes:
+   *          - Increment pending block number
+   *          - Store archive root for the new block number
+   *          - Store block metadata in circular storage (TempBlockLog)
+   *          - Update L1 gas fee oracle (when txs enabled)
+   *          - Consume inbox messages, insert outbox messages (when txs enabled)
```

Comment:
The inbox don't actually update state, but it is external call with reads. The outbox does ye 👍